devito/ops/transformer.py

```diff
@@ -37,7 +42,44 @@ def opsit(trees, count):
         parameters
     )
 
-    return ops_kernel
+    stencil_arrays_initializations = itertools.chain(*[
+        to_ops_stencil(p, node_factory.ops_args_accesses[p])
+        for p in parameters if isinstance(p, OpsAccessible)
+    ])
+
+    pre_time_loop = stencil_arrays_initializations
+
+    return pre_time_loop, ops_kernel
+
+
+def to_ops_stencil(param, accesses):
+    dims = len(accesses[0])
+    pts = len(accesses)
+    stencil_name = "s%sd_%s_%spt" % (dims, param.name, pts)
+
+    stencil_array = SymbolicArray(
+        name=stencil_name,
+        dimensions=(pts * dims,),
+        dtype=np.int32
+    )
+
+    ops_stencil = OpsStencil(stencil_name.upper())
```

Comment:
@vincepandolfo as I said on slack, the idea is to have something generic to use for all declaration.
Thus, I've created OPSDeclaration when I was working on ops_decl_block.

Therefore, I think we should wait for the 885 to be merged with the master. Then, use OPSDeclarator instead of OpsStencil.

This is to avoid having several classes that do the same thing.

Reply:
That doesn't work as each Ops type needs to specify its type

Reply:
Unless we pass the type name in the constructor, but I'm not a fan of that

Reply:
ops_stencil = OPSDeclarator(stencil_name.upper(), namespace['ops_stencil_type'])

Reply:
LocalObject receives name and type in its constructor. It is already a standard used.